<?xml version="1.0" ?>
<launch>

    <arg name="camera_name" default="None" />

    <arg name="publish_rect_rgb" default="false" />
    <arg name="publish_raw_rgb" default="false" />
    <arg name="publish_depth" default="false" />
    <arg name="publish_pointcloud" default="false" />
    <arg name="publish_camera_info" default="false" />
    <arg name="flip_x" default="false" />
    <arg name="flip_y" default="false" />
    
    <arg name="fps" default="30" />
    <arg name="image_width" default="1856" />
    <arg name="image_height" default="800" />
    <arg name="ros_rate" default="30" />

    <!-- Unitree Camera node -->
    <node name="camera_$(arg camera_name)" pkg="ias_unitree_camera" type="camera_node" args="$(arg camera_name)" respawn="false" output="screen">
        <param name="publish_rect_rgb" value="$(arg publish_rect_rgb)" />
        <param name="publish_raw_rgb" value="$(arg publish_raw_rgb)" />
        <param name="publish_depth" value="$(arg publish_depth)" />
        <param name="publish_camera_info" value="$(arg publish_camera_info)" />
        <param name="publish_pointcloud" value="$(arg publish_pointcloud)" />
        <param name="fps" value="$(arg fps)" />
        <param name="image_width" value="$(arg image_width)" />
        <param name="image_height" value="$(arg image_height)" />
        <param name="ros_rate" value="$(arg ros_rate)" />
        <param name="flip_x" value="$(arg flip_x)" />
        <param name="flip_y" value="$(arg flip_y)" />
    </node>

</launch>
